test/runner/lib/cloud/hwc.py

```diff
@@ -0,0 +1,70 @@
+# 2018.7.27 --- replace GCP to HWC
```

Comment:
Hi mattclay. Thank you very much for your detailed comments. Actually I don't quite understand how does the cloud test plugin work. Is there any documents about cloud test plugin? I didn't find that on website of Ansible. 
Regarding to the test account,  we have one. It would be great to set the CI system for Huawei cloud modules. The question is how to set it? Thanks again for your warm help!

Reply:
Unfortunately there isn't any documentation yet on creating cloud plugins. For plugins which can work with a docker container, the best thing to do is look at the existing plugins that use docker containers and ask questions here or on IRC.

For plugins that require external services, we'll need to discuss further to see if they support our requirements for use in CI.

One approach we take is to have permanent credentials for the service in our internal CI system. That system can then provision temporary credentials to provide to our public CI system for each test run. When a test run completes our internal CI system revokes the temporary test credentials and removes any resources created by the tests.

This of course requires the service to be able to provide several things:

- A master set of credentials capable of managing limited credentials for each test run.
- A way to remove all resources created during a test run without cooperation from the tests.

If this sounds like something you could support, then we'll need to discuss how to set this up in our private CI system.

Another option, similar to using a docker simulator, is if there is a simulator or sandbox that can run in a virtual machine on AWS or Azure. I'm guessing that probably doesn't apply for your use case.

Reply:
@mattclay Hi, thanks for your detailed reply. We already added  "Creating a setup_hwc/defaults/main.yml file with the necessary variables.
Adding the following to hwc_network_vpc/meta/main.yml" for huawei cloud provider. 

I will send you a master set of credentials in IRC private chat channel. I already send a message to you in IRC. Could you reply me when you have free time. Thank you.